core/src/main/java/org/libdohj/core/AltcoinNetworkParameters.java

```diff
@@ -31,6 +31,34 @@
      */
     Sha256Hash getBlockDifficultyHash(Block block);
 
+    /**
+     * Most coins use SHA256D for block hashes,
+     * but some do not.  If a coin uses SHA256D, then isBlockHashSHA256D should
+     * return true and calculateBlockHash should return null;
+     */
+
+    /**
+     *
+     * @return true if the coin uses SHA256D like bitcoin, false if it uses a different hash function
+     */
+    boolean isBlockHashSHA256D();
```

Comment:
this project is java 8 so you could use default implementations in the interface so subclasses don't have to implement these